drivers/mspi/CMakeLists.txt

```diff
@@ -7,3 +7,12 @@ zephyr_library_sources_ifdef(CONFIG_MSPI_AMBIQ_AP3   mspi_ambiq_ap3.c)
 zephyr_library_sources_ifdef(CONFIG_MSPI_AMBIQ_AP5   mspi_ambiq_ap5.c)
 zephyr_library_sources_ifdef(CONFIG_MSPI_DW          mspi_dw.c)
 zephyr_library_sources_ifdef(CONFIG_MSPI_EMUL        mspi_emul.c)
+zephyr_library_sources_ifdef(CONFIG_MSPI_ESP32       mspi_esp32.c)
+
+zephyr_sources_ifdef(
+    CONFIG_MSPI_ESP32
```

Comment:
This does not look right, you should not add hal vendor source files on the driver implementation, instead only the zephyr driver for a particular API should reside this, maybe you need to submit a PR to the hal_espressif if you need these files.

Reply:
Got it, will make separate PR for it.

Reply:
Done. [PR for it.](https://github.com/zephyrproject-rtos/hal_espressif/pull/471)